.github/workflows/status-checks.yml

```diff
@@ -0,0 +1,74 @@
+# Workflow to check whether changes to master fulfill all requirements.
+name: Status checks
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Build
+        run: go build -v -o irma-linux-amd64 ./irma
+
+  # We only check the build stage of the Dockerfile, because building the full Dockerfile requires
+  # downloading IRMA schemes. We check the status of the full Dockerfile once every week on master
+  # in the weekly-master-checks.
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Run Dockerfile build stage
+      run: docker build -t privacybydesign/irma:build --target build .
+
+    - name: Test Docker image
+      # Because we have only run the build stage, we have to explicitly set irma as entrypoint.
+      run: docker run --entrypoint irma privacybydesign/irma:build version
+      
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+
+    - name: Run gofmt
+      run: gofmt -d -e .
```

Comment:
Go report card uses the simplified option of `gofmt`. Maybe we want to use this option too. Then we would have some issues to fix.
https://goreportcard.com/report/github.com/privacybydesign/irmago#gofmt

Reply:
I suggest doing this after the session test refactor has been done, to avoid a merge conflict.